modules/BiomesAPI/build.gradle

```diff
@@ -35,17 +25,25 @@ buildscript {
     }
 }
 
-// Handle some logic related to where what is
-sourceSets {
-    main.java.outputDir = new File("$buildDir/classes")
-    test.java.outputDir = new File("$buildDir/testClasses")
-}
-JavaPluginConvention convention = project.getConvention().getPlugin(JavaPluginConvention.class);
-SourceSet mainSourceSet = convention.getSourceSets().getByName("main");
+import groovy.json.JsonSlurper
+import org.reflections.Reflections
+import org.reflections.util.FilterBuilder
+import org.reflections.scanners.SubTypesScanner
+import org.reflections.scanners.TypeAnnotationsScanner
+import org.reflections.util.ConfigurationBuilder
 
 ext {
     // Read environment variables, including variables passed by jenkins continuous integration server
     env = System.getenv()
+
+    // This is a fun one ... when versions switched to dynamic -SNAPSHOT or not based on branch existing modules using `master` would suddenly try publishing releases
+    // This won't work without additionally doing constant version bumps (perhaps via Git tags) - but too much work to switch around all modules at once
+    // Complicating things more the use of publish.gradle to centralize logic means modules and engine bits are treated the same, yet we need to vary modules
+    // Temporary workaround: default modules to bypass release management: master branch builds will still make snapshot builds for the snapshot repo
+    // If a module actually wants release management simply include `"isReleaseManaged" : true` in module.txt - this is needed for the engine repo embedded modules
```

Comment:
I don't like the fact that the generic `module.txt` now holds information specific to our build system. As it is not officially documented anywhere, I guess its not much of a deal. Nevertheless, I would like to get rid of it asap.

Reply:
Yep, it is a workaround for sure. Lets me get through this big release without an even bigger overhaul. It can be an undocced feature likely never to be used in module land as we can hopefully finish improving things soon.